src/components/address-search/AddressSearchInput.js

```diff
@@ -338,175 +275,35 @@ class AddressSearchInput extends Component<Props, State> {
         this.setState({isLoading: false});
         console.error(`Failed to fetch by keyword with error ${error}`);
       });
-  }
-
-  sortStreets = (a: Street, b: Street) => {
-    const aStreet = a.name[a.language] ? a.name[a.language].toLowerCase() : '';
-    const bStreet = b.name[b.language] ? b.name[b.language].toLowerCase() : '';
-
-    if(aStreet < bStreet) return -1;
-    if(aStreet > bStreet) return 1;
-    return 0;
-  }
-
-  searchStreets = debounce((input: string) => {
-    const fetchStreets = (language: 'fi' | 'sv') => {
-      const url = `${SERVICE_MAP_URL}/street/?${stringifyQuery({
-        page_size: 4,
-        input: input,
-        language: language,
-      })}`;
-      const request = new Request(url);
-
-      return fetch(request);
-    };
-
-    const fetchAllStreets = async() => {
-      const fiResponse = await fetchStreets('fi');
-      const svResponse = await fetchStreets('sv');
-
-      const fiResults = await fiResponse.json();
-      const svResults = await svResponse.json();
-
-      return [
-        ...fiResults.results.map((street) => ({...street, language: 'fi'})),
-        ...svResults.results.map((street) => ({...street, language: 'sv'})),
-      ];
-    };
-
-    this.setState({isLoading: true});
-
-    fetchAllStreets()
-      .then((results) => {
-        const streets = results.sort(this.sortStreets),
-          newState: any = {streets: streets};
-
-        if(streets.length === 1) {
-          const {selectedStreet} = this.state;
-
-          if(!selectedStreet || (streets[0].id !== selectedStreet.id)) {
-            newState.selectedStreet = streets[0];
-            this.searchAddresses(streets[0]);
-          }
-        } else {
-          newState.selectedStreet = null;
-          newState.addresses = [];
-        }
-
-        newState.isLoading = false;
-        this.setState(newState);
-      })
-      .catch((error) => {
-        this.setState({isLoading: false});
-        console.error(`Failed to fetch streets with error ${error}`);
-      });
-  }, 300);
+  }, 750)
```

Comment:
Magic number. Should use variable for this to give some context.

Is 750 milliseconds a bit too long debounce time? Would 500 ms (half a second) be better option? I know it is a tough balance between wait time and performing searches, but still this needs to be carefully decided. Waiting too long causes delays in filling to form, because user has to wait also the results and not only the debounce time.

Reply:
Yes, I experimented with this. I had to increase it from 300 anyway, because the search behaved a bit strange. But it seems to be fixed now. I set ti to 500ms and assigned it to variable.